MCDataProducts/inc/SimParticle.hh

```diff
@@ -269,34 +271,31 @@ namespace mu2e {
     art::Ptr<GenParticle>  _genParticle;
 
     // Information at the start of the track.
-    CLHEP::Hep3Vector       _startPosition;
-    CLHEP::HepLorentzVector _startMomentum;
-    double                  _startGlobalTime;
-    double                  _startProperTime;
+    XYZVectorD		    _startPosition;
+    XYZTVectorF             _startMomentum;
+    float		    _startGlobalTime;
+    float		    _startProperTime;
     unsigned                _startVolumeIndex;
     unsigned                _startG4Status;
     ProcessCode             _creationCode;
     IonDetail               _ion;
 
     // Information at the end of the track.
-    CLHEP::Hep3Vector       _endPosition;
-    CLHEP::HepLorentzVector _endMomentum;
-    double                  _endGlobalTime;
-    double                  _endProperTime;
+    XYZVectorD              _endPosition;
+    XYZTVectorF		    _endMomentum;
+    float                  _endGlobalTime;
```

Comment:
Are you sure storing times as floats is sufficient? Especially if one has to calculate differences of them?

Reply:
Note that all physical observables are created from StepPointMCs, which have double precision, so the only use case for this class is MC truth analysis.  What's the use case for that which needs double precision?  We couldn't come up with one in the meeting last week.

Reply:
6-7 significant digits for time seems small when tof can be measured with sub ns accuracy